Assignment4_FeatureEngineering.py

```diff
@@ -0,0 +1,260 @@
+        # Plot the figure
+        figure = px.scatter(x=df[column], y=y, trendline="ols")
+        figure.update_layout(
+            title=f"Variable: {column}: (t-value={t_value}) (p-value={p_value})",
+            xaxis_title=f"Variable: {column}",
+            yaxis_title="y",
+        )
+        figure.write_html(
+            file=f"assignment4/plots/ranking_{column}.html", include_plotlyjs="cdn"
+        )
+
+    cat_con = []
+    file = []
+    # mean_unweighted = []
+    # mean_weighted = []
+    for var in pred.columns:
+        if cat_or_bool(var) and response_var_type == "Boolean":
+            fpath = "assignment4/graph/cat_cat_heatmap" + var + ".html"
+
+            cm = confusion_matrix(var, df["response"])
+            plot = graph_objects.Figure(data=graph_objects.Heatmap(z=cm, zmax=cm.max()))
+
+            # Let's give out the titles and other details to our graph
+            plot.update_layout(
+                title="Categorical Predictor by Categorical Response ",
+                xaxis_title="Response",
+                yaxis_title="Predictor",
+            )
+
+            plot.write_html(
+                file=fpath,
+                include_plotlyjs="cdn",
+            )
+
+            cat_con.append("Categorical")
+            file.append("")
+
+        elif not cat_or_bool(var) and response_var_type == "Boolean":
+            fpath = "assignment4/graph/cat_con_dist" + var + ".html"
+            cat_con_dist(df, var, fpath)
+            cat_con.append("Continuous")
+            file.append("<a href=" + fpath + ">" + fpath + "</a>")
+
+        elif cat_or_bool(var) and response_var_type != "Boolean":
+            fpath = "assignment4/graph/con_cat_violin" + var + ".html"
+            cont_cat_violin(df, var, fpath)
+            cat_con.append("Categorical")
+            file.append("<a href=" + fpath + ">" + fpath + "</a>")
+
+        elif not cat_or_bool(var) and response_var_type != "Boolean":
+            fpath = "assignment4/graph/con_con_scatter" + var + ".html"
+
+            plot1 = df.plot(kind="scatter", x=var, y=df["response"])
+            plt.xlabel("Predictor")
+            plt.ylabel("Response")
+            plt.title("Continuous/Continuous")
+            plot1.write_html(
+                file=fpath,
+                include_plotlyjs="cdn",
+            )
+
+            cat_con.append("Continuous")
+            file.append("<a href=" + fpath + ">" + fpath + "</a>")
+
+        """
+         # Mean Squared Difference
+        mean_unweighted.append(mean_uw)
+        mean_weighted.append(mean_w)
+        """
+
+        # Assigning everything to a dataframe
+        result["cat_con"] = cat_con
+        result["t-value"] = t_value
+        result["p-value"] = p_value
+        result["m-plot"] = m_plot
+        result["graphs"] = file
+        result["Variable_importance"] = importance
+        # result['mean_unweighted']= mean_unweighted
```

Comment:
You did not do the calculation for difference of mean with response?

Reply:
I was facing difficulty doing it. Will update it asap.